### PR TITLE
Add Table Abstractions [WIP]

### PR DESCRIPTION
**Pull request by elas7**
Everything is working fine, but Flow types must be improved.

Features:
- Single `<Table>` element which takes data, configuration, and renders the formatted table.
- Each table component (`table`, `thead`, `tr`, etc) can be overriden with a custom component
- Each leaf component (`Header`, `Cell` & `Footer`) can be overriden for every column or
  for a specific column. They are called with row-level props and cell-level props, as appropriate.
- Each column can have an **accessor** string or custom function used to retrieve the right row's value for the column.

## Checklist

* [x] Feature developed
* [ ] Created/updated unit tests
* [ ] Comments in code
* [ ] Documentation written

### CODE REVIEW

**Comment by grgur** [comment]
@elas7 can you look at the conflicting files

File: app/components/BudgetGrid/index.js [hunk 1]
```diff
@@ -22,25 +29,60 @@ export class BudgetGrid extends React.Component<BudgetGridProps> {
...
-        </tbody>
+    const ResponsiveCell = ({ columnName, value, className }: ResponsiveCellProps) => (
+      <div className={`${styles.responsiveCell} ${className || ''}`}>
+        <div className={styles.cellLabel}>{columnName}</div>
+        <div className={styles.cellContent}>{value}</div>
+      </div>
+    );
+
+    const TableComponent = ({ children, ...otherProps }) => (
+      <table {...otherProps}>
```
Comment (grgur) [inline app/components/BudgetGrid/index.js]: is this intentionally `table` and not `Table`?

Comment (elas7) [reply app/components/BudgetGrid/index.js]: Yes, that' intentionally `table`. 
`TableComponent` here is overriding the default `TableComponent` used by `Table`.
We do this to render a custom table footer.

**Comment by elas7** [comment]
Conflicts are fixed

**Pull request closed without merging**